src/datatable/datatable.jl

```diff
@@ -757,8 +734,8 @@ end
 hcat!(dt::DataTable, x::CategoricalArray) = hcat!(dt, DataTable(Any[x]))
```

Comment:
These definitions are all the same, they should be merged into an `AbstractVector` method.